gdx/src/com/badlogic/gdx/utils/XmlReader.java

```diff
@@ -534,6 +538,14 @@ public void removeChild (Element child) {
 			if (children != null) children.removeValue(child, true);
 		}
 
+		public void replaceChild (Element child, Element replacement) {
+			if (children == null) throw new GdxRuntimeException("Element has no children: " + name);
+			final int index = children.indexOf(child, true);
+			if (index == -1) throw new GdxRuntimeException("Element does not contain child: " + child);
+			children.removeIndex(index);
+			children.insert(index, replacement);
```

Comment:
Is there any reason why this doesn't use `children.set(index, replacement);` ? Removing by index has to move later items toward the start, and then inserting it has to move them again toward the end. The appropriate method would be set(), I think, unless you want to only remove or only insert an item.

Reply:
You're absolutely right.